File: Java/BinaryTree.java
import java.util.Scanner;

class TreeNode {
    int data;
    TreeNode left;
    TreeNode right;

    TreeNode(int data) {
        this.data = data;
        this.left = null;
        this.right = null;
    }
}

class BinaryTree {
    TreeNode root;

    TreeNode insertNode(TreeNode node, int data) {
        if (node == null) {
            return new TreeNode(data);
        }

        if (data < node.data) {
            node.left = insertNode(node.left, data);
        } else if (data > node.data) {
            node.right = insertNode(node.right, data);
        }

        return node;
    }

    boolean searchNode(TreeNode node, int data) {
        if (node == null) {
            return false;
        }

        if (node.data == data) {
            return true;
        }

        if (data < node.data) {
            return searchNode(node.left, data);
        } else {
            return searchNode(node.right, data);
        }
    }

    void inorderTraversal(TreeNode node) {
        if (node != null) {
            inorderTraversal(node.left);
            System.out.print(node.data + " ");
            inorderTraversal(node.right);
        }
    }
}

public class Main {
    public static void main(String[] args) {
        BinaryTree binaryTree = new BinaryTree();
        Scanner scanner = new Scanner(System.in);
        int option;

        do {
            System.out.println("Binary Tree Options");
            System.out.println("1. Insert node");
            System.out.println("2. Search node");
            System.out.println("3. Inorder traversal");
            System.out.println("4. Exit");
            System.out.print("Enter your choice: ");
            option = scanner.nextInt();

            switch (option) {
                case 1:
                    System.out.print("Enter the value to insert: ");
                    int value = scanner.nextInt();
                    binaryTree.root = binaryTree.insertNode(binaryTree.root, value);
                    break;
                case 2:
                    System.out.print("Enter the value to search: ");
                    int searchValue = scanner.nextInt();
                    if (binaryTree.searchNode(binaryTree.root, searchValue)) {
                        System.out.println(searchValue + " found in the tree");
                    } else {
                        System.out.println(searchValue + " not found in the tree");
                    }
                    break;
                case 3:
                    System.out.println("Inorder Traversal:");
                    binaryTree.inorderTraversal(binaryTree.root);
                    System.out.println();
                    break;
                case 4:
                    System.out.println("Exiting...");
                    break;
                default:
                    System.out.println("Invalid option");
            }

            scanner.nextLine(); // Consumir el carácter de nueva línea restante

        } while (option != 4);
    }
}

File: Kotlin/BinaryTree.kt
import java.util.*

class TreeNode(var data: Int) {
    var left: TreeNode? = null
    var right: TreeNode? = null
}

class BinaryTree {
    var root: TreeNode? = null

    fun insertNode(data: Int) {
        root = insertNodeRec(root, data)
    }

    private fun insertNodeRec(node: TreeNode?, data: Int): TreeNode {
        if (node == null) {
            return TreeNode(data)
        }

        if (data < node.data) {
            node.left = insertNodeRec(node.left, data)
        } else if (data > node.data) {
            node.right = insertNodeRec(node.right, data)
        }

        return node
    }

    fun searchNode(data: Int): Boolean {
        return searchNodeRec(root, data)
    }

    private fun searchNodeRec(node: TreeNode?, data: Int): Boolean {
        if (node == null) {
            return false
        }

        if (node.data == data) {
            return true
        }

        return if (data < node.data) {
            searchNodeRec(node.left, data)
        } else {
            searchNodeRec(node.right, data)
        }
    }

    fun inorderTraversal() {
        inorderTraversalRec(root)
    }

    private fun inorderTraversalRec(node: TreeNode?) {
        if (node != null) {
            inorderTraversalRec(node.left)
            print("${node.data} ")
            inorderTraversalRec(node.right)
        }
    }
}

fun main() {
    val binaryTree = BinaryTree()
    val scanner = Scanner(System.`in`)
    var option: Int

    do {
        println("Binary Tree Options")
        println("1. Insert node")
        println("2. Search node")
        println("3. Inorder traversal")
        println("4. Exit")
        print("Enter your choice: ")
        option = scanner.nextInt()

        when (option) {
            1 -> {
                print("Enter the value to insert: ")
                val value = scanner.nextInt()
                binaryTree.insertNode(value)
            }
            2 -> {
                print("Enter the value to search: ")
                val value = scanner.nextInt()
                if (binaryTree.searchNode(value)) {
                    println("$value found in the tree")
                } else {
                    println("$value not found in the tree")
                }
            }
            3 -> {
                println("Inorder Traversal:")
                binaryTree.inorderTraversal()
                println()
            }
            4 -> println("Exiting...")
            else -> println("Invalid option")
        }

        scanner.nextLine() // Consumir el carácter de nueva línea restante

    } while (option != 4)
}
